### PR TITLE
minor fixes, in particular to job state display in systemctl

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/shared/bus-unit-util.c [hunk 1]
```diff
@@ -1127,6 +1127,9 @@ static int bus_job_get_service_result(BusWaitForJobs *d, char **result) {
         assert(d->name);
         assert(result);
 
+        if (!endswith(d->name, ".service"))
+                return -EINVAL;
```
Comment (sourcejedi) [inline src/shared/bus-unit-util.c:1131]: But `Result` is also available on socket, mount, automount, timer, swap, and path units?  (Also scopes, but I suspect that's  not relevant).

Comment (poettering) [reply src/shared/bus-unit-util.c:1131]: but the values live in a different namespace, even though they are very similar. Also, we query the one off the Service interface, not the others. That means, if we want to support the other results too, we'd have to add some more code anyway, to use the right dbus interface...

Comment (poettering) [reply src/shared/bus-unit-util.c:1131]: (adding support for other unit types should be left for a different PR. For now, we should at least fix that we request the Result property on the wrong interface for non-service units, which this patch does)

File: src/systemctl/systemctl.c [hunk 2]
```diff
@@ -3348,8 +3348,10 @@ static int logind_check_inhibitors(enum action a) {
                 if (!sv)
                         return log_oom();
 
-                if ((pid_t) pid < 0)
-                        return log_error_errno(ERANGE, "Bad PID %"PRIu32": %m", pid);
+                if (!pid_is_valid((pid_t) pid)) {
+                        log_error("Bad PID %"PRIu32": %m", pid);
```
Comment (sourcejedi) [inline src/systemctl/systemctl.c]: `pid_is_valid()` doesn't set `errno`.  Previous version was correct, I think.

File: src/shared/bus-unit-util.c (hunk 1 above)
Comment (sourcejedi) [reply src/shared/bus-unit-util.c:1131]: Sure, that makes sense.

File: src/systemctl/systemctl.c (hunk 2 above)
Comment (poettering) [reply src/systemctl/systemctl.c]: ouch, indeed

Comment (poettering) [reply src/systemctl/systemctl.c]: (well, it shows ERANGE, pulling this error out of thin air... we shouldn't do that really, there's no ERANGE received here anywhere, hence we shouldn't confuse the user seeing this with it)

**Comment by poettering** [comment]
force pushed a new version with @sourcejedi's comments fixed

**Comment by sourcejedi** [comment]
OK.  (I then noticed the pid range error might be less confusing if we fudged it and printed it as signed.  But it's really a debugging message, it's not expected to ever happen).

**Comment by poettering** [comment]
I'll take your "OK" as positive review and merge... ;-)

CI failure appears unrelated

**Pull request merged**